### DISCUSSION TITLE
Questions about some json attributes

### DISCUSSION THREAD

**Discussion by baptiste** [excalidraw/excalidraw#2672] [Q&A] [answered]
I've been playing with excalidraw files generated from R; the json format is generally straight-forward but there are some fields that I couldn't work out:

- “boundElementIds”: []
- “version”: 160
- “versionNonce”: 1571380943

I'm also curious to know the difference between "draw" and "line" elements.

Thanks for any clarifications you can provide!

**Comment by dwelle** [comment, marked as answer]
Good questions. I've made a PR to add TSDoc for these (and more): https://github.com/excalidraw/excalidraw/pull/2673

- `boundElementIds` — ids of (linear) elements that are bound to this element
- `version` — integer that is sequentially incremented on each change. Currently used to reconcile elements during collaboration or when saving to server.
- `versionNonce` — random integer that is regenerated on each change. Used for deterministic reconciliation of updates during collaboration, in case the `version`s are identical.

> I'm also curious to know the difference between "draw" and "line" elements.

From a JSON schema standpoint, I think there's none AFAIK. So there's only behavior difference.

**Reply by baptiste** [reply]
Great, thanks for the clarifications!

Here's R's volcano level lines

![Untitled-2020-12-26-0856-3-1](https://user-images.githubusercontent.com/18970/103160814-e2310380-483d-11eb-995b-09ba7070be7b.png)
